recipes/mp-units/0.7.0/conanfile.py

```diff
@@ -1,70 +1,110 @@
-from conans import ConanFile, CMake, tools
-from conans.tools import Version, check_min_cppstd
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.59.0"
+
 
 class MPUnitsConan(ConanFile):
     name = "mp-units"
     homepage = "https://github.com/mpusz/units"
-    description = "Physical Units library for C++"
-    topics = ("units", "dimensions", "quantities", "dimensional-analysis", "physical-quantities", "physical-units", "system-of-units", "cpp23", "cpp20", "library", "quantity-manipulation")
+    description = "Physical Quantities and Units library for C++"
+    topics = (
+        "units",
+        "dimensions",
+        "quantities",
+        "dimensional-analysis",
+        "physical-quantities",
+        "physical-units",
+        "system-of-units",
+        "system-of-quantities",
+        "isq",
+        "si",
+        "library",
+        "quantity-manipulation",
+    )
     license = "MIT"
     url = "https://github.com/conan-io/conan-center-index"
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake_find_package_multi"
+    package_type = "header-library"
     no_copy_source = True
+    generators = "CMakeToolchain", "CMakeDeps"
 
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return "20"
 
-    def requirements(self):
+    @property
+    def _minimum_compilers_version(self):
+        return {"gcc": "10.3", "clang": "12", "apple-clang": "13", "msvc": "192"}
+
+    @property
+    def _use_range_v3(self):
         compiler = self.settings.compiler
-        self.requires("fmt/7.1.3")
+        version = Version(self.settings.compiler.version)
+        return "clang" in compiler and compiler.libcxx == "libc++" and version < 14
+
+    def requirements(self):
         self.requires("gsl-lite/0.38.0")
-        if compiler == "clang" and compiler.libcxx == "libc++":
+        self.requires("fmt/7.1.3")
+        if self._use_range_v3:
             self.requires("range-v3/0.11.0")
```

Comment:
For the Conan V1 pipeline to pass, I think we need to add `transitive_headers=True` for at least `range-v3` according to the latest error, possibly for the other two dependencies, as well. Conan V1 doesn't use `package_type` to figure this out for us, unfortunately. That's why the V2 pipeline passes.

```suggestion
            self.requires("range-v3/0.11.0", transitive_headers=True)
```

Reply:
It seems that was not the source of the issue :-( I do not know why it works on Conan 2.0 and does not work on 1.X :-(

Reply:
I'll look into this a bit more, and see if I can figure out what's going on.